### PR TITLE
rotki: init at 1.31.2

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Inits privacy friendly portfolio manager.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Lurkki14** [comment]
Adding yourself to maintainers needs to be a separate commit

**Comment by adamjedrzejewski** [comment]
@IogaMaster

**Comment by IogaMaster** [comment]
```sh
cannot execute /nix/store/2pwssllbbyjrkcgqq92x7kmswlzs5dyz-rotki-1.31.2/bin/rotki: You are trying to run an unpatched binary on nixos, but you have not configured NIX_LD or NIX_LD_x86_64-linux. See https://github.com/Mic92/nix-ld for more details
```
Current issue

**Review by IogaMaster — CHANGES_REQUESTED** [review]
Due to the issue I mentioned I cannot approve this pr.

**Comment by shazow** [comment]
I was just looking for a nix packaging of rotki, thanks for working on this! I also stumbled on this one, in case it's a useful reference: https://github.com/FlorianFranzen/nixrc/blob/master/pkgs/rotki.nix

**Comment by adamjedrzejewski** [comment]
@IogaMaster: updated, I tested it on a fresh NixOS install in a VM, check if it works for you

**Review by IogaMaster — APPROVED** [review]
Works great!
Thanks for contributing to nixpkgs! :smile:

File: pkgs/by-name/ro/rotki/package.nix
Comment (Janik-Haag) [inline pkgs/by-name/ro/rotki/package.nix:68]: to be honest this is quite ugly to be honest and it would be preferable to build it from source, this would probably also remove the platform constraints.

Comment (IogaMaster) [reply pkgs/by-name/ro/rotki/package.nix:68]: Can agree

**Pull request closed without merging**
